crates/lib3h/src/transport/memory_mock/transport_memory.rs

```diff
@@ -233,53 +271,75 @@ impl Transport for TransportMemory {
                 outbox.append(&mut output);
             }
         }
-        // Process my Servers
-        let mut to_connect_list: Vec<Url> = Vec::new();
-        for server_uri in &self.my_servers {
+        // Process my Servers: process IncomingConnectionEstablished first
+        let mut to_connect_list: Vec<(Url, ConnectionId)> = Vec::new();
+        let mut output = Vec::new();
+        for my_server_uri in &self.my_servers {
             let server_map = memory_server::MEMORY_SERVER_MAP.read().unwrap();
-            let server = server_map.get(server_uri).expect("My server should exist.");
-            let (success, output) = server.lock().unwrap().process()?;
+            let mut my_server = server_map
+                .get(my_server_uri)
+                .expect("My server should exist.")
+                .lock()
+                .unwrap();
```

Comment:
Do you think an `expect` could be useful here, or it's meaningless after a lock?